lib/java.gradle

```diff
@@ -53,12 +53,17 @@ configure(projectsWithFlags('java')) {
         }
     }
 
-    // Add the generated source directories to the source sets.
-    project.sourceSets.all { sourceSet ->
-        sourceSet.java.srcDir file("${project.ext.genSrcDir}/${sourceSet.name}/java")
-        sourceSet.resources.srcDir file("${project.ext.genSrcDir}/${sourceSet.name}/resources")
-    }
     afterEvaluate {
+        // Add the generated source directories to the source sets.
+        // This should run in 'afterEvaluate' so that protobuf plugin which also adds
+        // generated files to java source set applied first.
+        project.sourceSets.all { sourceSet ->
+            def javaSrcDir = file("${project.ext.genSrcDir}/${sourceSet.name}/java")
+            if (!sourceSet.java.srcDirs.contains(javaSrcDir)) {
+                sourceSet.java.srcDir javaSrcDir
+            }
+            sourceSet.resources.srcDir file("${project.ext.genSrcDir}/${sourceSet.name}/resources")
```

Comment:
How about also applying the same approach for this?
We might add xRPC later and the plugin of the xRPC could probably add this directory which might cause the same error.